src/playercards/AllCardsBag.ttslua

```diff
@@ -131,70 +254,179 @@ function buildSupplementalIndexes()
-          if cycleIndex[cycleName] == nil then
-            cycleIndex[cycleName] = { }
+        end
+      end
+
+      if cardMetadata.willpowerIcons then
+        if not iconIndex.willpower[cardMetadata.willpowerIcons] then
+          iconIndex.willpower[cardMetadata.willpowerIcons] = { }
+        end
+        table.insert(iconIndex.willpower[cardMetadata.willpowerIcons], cardMetadata.id)
+      else
+        table.insert(iconIndex.willpower[0], cardMetadata.id)
+      end
+      if cardMetadata.intellectIcons then
+        if not iconIndex.intellect[cardMetadata.intellectIcons] then
+          iconIndex.intellect[cardMetadata.intellectIcons] = { }
+        end
+        table.insert(iconIndex.intellect[cardMetadata.intellectIcons], cardMetadata.id)
+      else
+        table.insert(iconIndex.intellect[0], cardMetadata.id)
+      end
+      if cardMetadata.agilityIcons then
+        if not iconIndex.agility[cardMetadata.agilityIcons] then
+          iconIndex.agility[cardMetadata.agilityIcons] = { }
+        end
+        table.insert(iconIndex.agility[cardMetadata.agilityIcons], cardMetadata.id)
+      else
+        table.insert(iconIndex.agility[0], cardMetadata.id)
+      end
+      if cardMetadata.combatIcons then
+        if not iconIndex.combat[cardMetadata.combatIcons] then
+          iconIndex.combat[cardMetadata.combatIcons] = { }
+        end
+        table.insert(iconIndex.combat[cardMetadata.combatIcons], cardMetadata.id)
+      else
+        table.insert(iconIndex.combat[0], cardMetadata.id)
+      end
+      if cardMetadata.wildIcons then
+        if not iconIndex.wild[cardMetadata.wildIcons] then
+          iconIndex.wild[cardMetadata.wildIcons] = { }
+        end
+        table.insert(iconIndex.wild[cardMetadata.wildIcons], cardMetadata.id)
+      else
+        table.insert(iconIndex.wild[0], cardMetadata.id)
+      end
+
+      if cardMetadata.type then
+        if not typeIndex[string.lower(cardMetadata.type)] then
+          typeIndex[string.lower(cardMetadata.type)] = { }
+        end
+        table.insert(typeIndex[string.lower(cardMetadata.type)], cardMetadata.id)
+      end
+
+      if cardMetadata.cost then
+        if not costIndex[cardMetadata.cost] then
+          costIndex[cardMetadata.cost] = { }
+        end
+        table.insert(costIndex[cardMetadata.cost], cardMetadata.id)
+      end
+
+      if cardMetadata.permanent then
+        table.insert(permanentIndex, cardMetadata.id)
+      end
+
+      if cardMetadata.uses then
+        local usesCount = cardMetadata.uses[1].count or -1
+        if not usesIndex[usesCount] then
+          usesIndex[usesCount] = { }
+        end
+        table.insert(usesIndex[usesCount], cardMetadata.id)
+        if not usesIndex[string.lower(cardMetadata.uses[1].type)] then
+          usesIndex[string.lower(cardMetadata.uses[1].type)] = { }
+        end
+        table.insert(usesIndex[string.lower(cardMetadata.uses[1].type)], cardMetadata.id)
+      end
+      if bonded then
```

Comment:
This is a repetition of LOC 314: https://github.com/argonui/SCED/pull/356/files#diff-5025e54fb68163d3f0c0619249100efca58320292451f20d76c5250e2d27d2deR314